packages/alchemy/src/middleware/simulate-uo.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  deepHexlify,
+  resolveProperties,
+  type UserOperationStruct,
+} from "@alchemy/aa-core";
+import type { AlchemyProvider } from "../provider/base.js";
+import type { ClientWithAlchemyMethods } from "./client.js";
+
+export const withAlchemyUserOpSimulation = <P extends AlchemyProvider>(
```

Comment:
should this be part of enhanced api actions?

Reply:
no let's keep this method with the other alchemy AA enhanced apis.

the enhanced apis are coming from the alchemy-sdk (ethers.js backed), which requires a peer dependency and is a different set.